yrs/src/block.rs

```diff
@@ -713,16 +714,83 @@ impl ItemPosition {
 }
 
 /// Bit flag (4th bit) for a marked item - not used atm.
-const ITEM_FLAG_MARKED: u8 = 0b1000;
+const ITEM_FLAG_MARKED: u8 = 0b0000_1000;
 
 /// Bit flag (3rd bit) for a tombstoned (deleted) item.
-const ITEM_FLAG_DELETED: u8 = 0b0100;
+const ITEM_FLAG_DELETED: u8 = 0b0000_0100;
 
 /// Bit flag (2nd bit) for an item, which contents are considered countable.
-const ITEM_FLAG_COUNTABLE: u8 = 0b0010;
+const ITEM_FLAG_COUNTABLE: u8 = 0b0000_0010;
 
 /// Bit flag (1st bit) used for an item which should be kept - not used atm.
-const ITEM_FLAG_KEEP: u8 = 0b0001;
+const ITEM_FLAG_KEEP: u8 = 0b0000_0001;
+
+#[repr(transparent)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct ItemFlags(u8);
```

Comment:
Initially I wanted to move information about string encoding onto bits of `Item::info`, but it wouldn't solve an issue (without further modifications). One good thing, that came out of it, was to change bit flags into their own dedicated type, which now is more safe and gives us explicit knowledge about which bits are used and how. It's safer and easier to debug. Also doesn't introduce any performance overhead (I've checked that).